index.js

```diff
@@ -21,10 +27,8 @@ module.exports = (input, needle, replacement) => {
 		}
 
 		matchCount++;
-
 		const replaceStr = typeof replacement === 'string' ? replacement : replacement(needle, matchCount, input);
-
-		ret += input.slice(prevIndex, index) + replaceStr;
+		ret += (prevIndex === opts.fromIndex) ? input.slice(0, index) + replaceStr : input.slice(prevIndex, index) + replaceStr;
```

Comment:
This logic doesn't make much sense to me. The first case is only ever true when `prevIndex === opts.fromIndex`, which would only be the case when `index + needle.length` happens to be exactly the same value.

If it is indeed correct, can you explain your thinking? Or better yet, make the code itself clearer.